pkg/compiler/compiler_test.go

```diff
@@ -7,30 +7,45 @@ import (
 	"testing"
 
 	"github.com/CityOfZion/neo-go/pkg/compiler"
+	"github.com/stretchr/testify/require"
 )
 
 const examplePath = "../../examples"
 
-func TestExamplesFolder(t *testing.T) {
-	infos, err := ioutil.ReadDir(examplePath)
-	if err != nil {
-		t.Fatal(err)
-	}
+type testCase struct {
+	name     string
+	function func(string, string)
+}
 
-	for _, info := range infos {
-		infos, err := ioutil.ReadDir(path.Join(examplePath, info.Name()))
-		if err != nil {
-			t.Fatal(err)
-		}
-		if len(infos) == 0 {
-			t.Fatal("detected smart contract folder with no contract in it")
-		}
+func TestCompiler(t *testing.T) {
+	testCases := []testCase{
+		{
+			name: "TestCompile",
+			function: func(targetPath string, fileName string) {
+				if err := compileFile(targetPath); err != nil {
+					t.Fatal(err)
+				}
+			},
+		},
+		{
+			name: "TestCompileAnsSave",
+			function: func(targetPath string, fileName string) {
+				outfile := fileName + ".avm"
+				if _, err := compiler.CompileAndSave(targetPath, &compiler.Options{Outfile: outfile}); err != nil {
```

Comment:
What will it do if there is already an avm file in the directory? Overwrite? It's too dangerous, we're unit-testing compiler and writing things into the `examples` two directories above. I think we can test it with a dummy "hello world" contract in the `testdata`, there is no need to compile everything we have in examples.

Reply:
1. `testdata` : do you mean that it's enough to create there a .go contract like Hello World and use it in both tests (compile and compileAndSave)?
2. having overwritten .avm file is not ok, but I can't imagine situation when it could be critical. In the worst case scenario you will have changed file/failed unit test. But we're not going to put .avm files there. Anyway I can keep them in separate dir (but what if someone will put exactly in this test repo some .avm file too?)

Reply:
It's OK to compile things from `examples`, it's not to save the results in the `examples` directory. So HW in `testdata` to test `CompileAndSave` should be fine (we're mostly testing `AndSave` part here, codegeneration is exactly the same.